File: app/src/main/java/com/bzh/dytt/BaseActivity.java
package com.bzh.dytt;


import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import com.umeng.analytics.MobclickAgent;

public class BaseActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        doCreate();
    }

    @Override
    protected void onResume() {
        super.onResume();
        MobclickAgent.onResume(this);
        doResume();
    }

    @Override
    protected void onPause() {
        doPause();
        MobclickAgent.onPause(this);
        super.onPause();
    }

    protected void doCreate(){}

    protected void doResume(){}

    protected void doPause(){}
}

File: app/src/main/java/com/bzh/dytt/BaseActivity.kt
package com.bzh.dytt

import android.annotation.SuppressLint
import android.support.v7.app.AppCompatActivity
import com.umeng.analytics.MobclickAgent

@SuppressLint("Registered")
open class BaseActivity : AppCompatActivity() {

    override fun onResume() {
        super.onResume()
        MobclickAgent.onResume(this)
    }

    override fun onPause() {
        MobclickAgent.onPause(this)
        super.onPause()
    }
}